website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx

```diff
@@ -0,0 +1,140 @@
+---
+layout: docs
+page_title: Upgrading to 1.10.0
+description: >-
+  Specific versions of Consul may have additional information about the upgrade
+  process beyond the standard flow.
+---
+
+# Upgrading to 1.10.0
+
+<EnterpriseAlert />
+
+## Introduction
+
+This guide explains how to best upgrade a single Consul Enterprise datacenter to v1.10.0
+from a version of Consul that is forwards compatible with v1.10. If you are on a major 
+version of Consul prior to 1.8, please follow the link for the version you are on from 
+[here](/docs/upgrading/instructions). If you are already on major versions 1.8 or 1.9
+this guide will go over the procedures required for upgrading to v1.10. If you have
+multiple Consul datacenters then they are upgraded in the normal order with each
+upgrade taking into account the instructions below.
+
+For the open source version of Consul please follow the 
+[General Upgrade Process](/docs/upgrading/instructions/general-process).
+
+~> You can only upgrade to Consul Enterprise 1.10 from a version of Consul Enterprise 
+1.8 >= 1.8.13 or 1.9 >= 1.9.7. Other versions of Consul Enterprise are not forwards
+compatible with v1.10 and will cause issues during the upgrade that could result in
+agents failing to start.
+
+## Requirements
+
+- All Consul servers, clients and snapshot agents should be on a version of Consul >= 1.8.0 and < 1.10.0.
+
+## Assumptions
+
+This guides makes the following assumptions:
+
+- You are familiar with the [General Upgrade Process](/docs/upgrading/instructions/general-process).
+- You have the ability to run Consul CLI commands. 
+
+## Considerations
+
+The main breaking changes in Consul Enterprise 1.10 that needs special handling during upgrading are
+the licensing changes outlined on the [Specific Version Details](/docs/upgrading/upgrade-specific#consul-1-10-0)
+page. There are notes on that paged about other changes that might cause issues during an upgrade as well.
+You can find more granular details in the full [changelog](https://github.com/hashicorp/consul/blob/release/release/1.10.0/CHANGELOG.md#1100-june-22-2021).
+Looking through these changes prior to upgrading is highly recommended.
+
+## Procedures
+
+**1.** Retrieve the datacenter's current license by executing the following CLI command:
+
+```shell
+consul license get -signed
+```
+
+This will be useful to have on hand when preparing the license for the 1.10 upgrades later
+on in the process.
+
+**2.** Determine if ACLs are enabled by running the following CLI command.
+
+```shell
+consul info | grep "acl ="
+````
+
+If you receive output like either of the following then ACLs are enabled:
+
+```text
+Error querying agent: Unexpected response code: 403 (Permission denied)
+```
+
+```text
+        acl = enabled
+
+```
+
+<Tabs>
+<Tab heading="ACLs Enabled">
```

Comment:
I like the tabs approach which limits the apparent complexity. Perhaps we should make double sure this is easy to follow with a sentence above like:

> Select the correct tab below to see the steps based on whether you have ACLs enabled of disabled in your environment.

Reply:
Does it make sense to have a separate tab for Kubernetes? The upgrade path for Kubernetes will read slightly differently and I can either add it here as a suggestion or branch off of this PR branch and document it there.

Reply:
We wouldnt need a separate tab for ACLs enabled vs disabled for Kubernetes as the helm chart manages that complexity under the hood for the user. They only need to provide us with the license file and we ensure we mount it to the right components depending on their existing configuration.

Reply:
> They only need to provide us with the license file and we ensure we mount it to the right components depending on their existing configuration.

In fact users will already have a license file in a secret so they only need to bump the Consul image and then run `helm upgrade`. Basically we should copy this text: https://www.consul.io/docs/enterprise/license/faq#kubernetes somewhere in here.

Reply:
@lkysow We created a tab for Kubernetes and have described just that. We could add the faq steps there as well. The only additional thing we added were steps to ensure people who had consul servers external to kubernetes had steps that they could follow as well. https://consul-m514xwknh-hashicorp.vercel.app/docs/upgrading/instructions/upgrade-to-1-10-x

Reply:
perfect